src/router/metadata/metadataHandler.ts

```diff
@@ -72,7 +72,16 @@ export default class MetadataHandler implements Capabilities {
         const generatedResources = await this.generateResources(request.fhirVersion);
         const security = makeSecurity(auth, this.hasCORSEnabled);
         const rest = makeRest(generatedResources, security, profile.systemOperations, !!profile.bulkDataAccess);
-        const capStatement = makeStatement(rest, productInfo, server.url, request.fhirVersion);
+
+        // determine if our implentation.url is a static tld or dynamic
+        let serverUrl = server.url;
+        if (server.dynamicHostName && !isUndefined(request.hostName) && !isNull(request.hostName)) {
+            // use a dynamic impelmentation.url so clients to each TLD have dynamic cannonical URLs
+            const parsedUrl = new URL(server.url);
+            parsedUrl.hostname = request.hostName;
+            serverUrl = parsedUrl.href;
+        }
+        const capStatement = makeStatement(rest, productInfo, serverUrl, request.fhirVersion);
```

Comment:
I think that this code should only exist on `setServerUrl.ts`. Ideally that's the only place where code to determine the server url runs and and all other components just read from `res.locals.serverUrl`

I suggest updating `CapabilitiesRequest` to have a `fhirServiceBaseUrl` field instead of `hostname` and then update `metadataRoute.ts` to read `res.locals.serverUrl` and pass it as a parameter
https://github.com/awslabs/fhir-works-on-aws-routing/blob/5ef50dad0c1f08fdf7ec88bb04b5584626b7f042/src/router/routes/metadataRoute.ts#L37

Reply:
Updated interface in this push, https://github.com/awslabs/fhir-works-on-aws-interface/pull/96/commits/110ae4341bcd975a2e2e75e3543deb4478459917 and updated the metadata express route and handler in this push https://github.com/awslabs/fhir-works-on-aws-routing/pull/153/commits/500cdc52b81e5c03a37fa67957dbbf3f96349c22